iocore/net/SSLNetVConnection.cc

```diff
@@ -1080,7 +1096,7 @@ SSLNetVConnection::sslServerHandShakeEvent(int &err)
     SSLDebugVC(this, "SSL handshake error: %s (%d), errno=%d", SSLErrorName(ssl_error), ssl_error, err);
 
     // start a blind tunnel if tr-pass is set and data does not look like ClientHello
-    char *buf = handShakeBuffer->buf();
+    char *buf = handShakeBuffer ? handShakeBuffer->buf() : NULL;
```

Comment:
This is not needed. Inside the sslServerHandShakeEvent(), the handshake is stills not completed and the handShakeReader is not freed by free_handshake_buffers().

Just keeps the original code here.